lib/rules/function-calc-no-unspaced-operator/index.cjs

```diff
@@ -65,169 +62,301 @@ const rule = (primary, _secondaryOptions, context) => {
 			let needsFix = false;
 			const valueIndex = declarationValueIndex(decl);
 
-			/**
-			 * @param {import('@csstools/css-parser-algorithms').TokenNode} operatorNode
-			 * @param {import('@csstools/css-parser-algorithms').ContainerNode} parent
-			 * @param {number} index
-			 * @param {'before' | 'after'} position
-			 * @returns {void}
-			 */
-			function checkWhitespaceAroundOperator(operatorNode, parent, index, position) {
-				const isBefore = position === 'before';
-				const aroundNode = parent.value[index + (isBefore ? -1 : 1)];
+			const tokens = cssTokenizer.tokenize({ css: value });
 
-				if (cssParserAlgorithms.isWhitespaceNode(aroundNode)) {
-					const whitespace = getWhitespace(aroundNode);
+			{
+				// Step 1
+				// Step 1.1
+				// Re-tokenize dimensions with units containing dashes.
+				// These might be typo's.
+				// For example: `10px-20px` has a unit of `px-20px`
+				for (let i = 0; i < tokens.length; i++) {
+					const token = tokens[i];
 
-					if (whitespace === ' ') return;
+					// NOTE:
+					// This is a mess, I am fairly certain that it is correct, but too much is happening with strings.
+					// This will be brittle and might break when the css-tokenizer is updated.
+					// It will also produce invalid CSS for exotic units, nu such units exists today.
```

Comment:
typo

nu >  no

<hr/>

question

> nu such units exists today

There's `rpx` which is a weapp unit.
e.g.
```
    'unit-no-unknown': [
      true,
      {
        ignoreUnits: ['rpx'],
      },
    ],
```

Should we add a test?

Reply:
This is an exotic unit : `1🧛🏼`

```
'1🧛🏼'.length // 5
```

However, it seems that I did end up wiring everything up correctly and these just work.
I will clarify the comments and add tests

Reply:
And this is also an exotic unit `1\\23`.
This is not working correctly.

But again, no such units exist today.